src/script/interpreter.cpp

```diff
@@ -304,7 +308,7 @@ bool EvalScript(vector<vector<unsigned char> >& stack, const CScript& script, un
                 opcode == OP_DIV ||
                 opcode == OP_MOD ||
                 opcode == OP_LSHIFT ||
-                opcode == OP_RSHIFT)
+                opcode == OP_RSHIFT))
```

Comment:
I would assume all of these would be treated as "unknowns" and would immediately return `set_success(serror)` or an error if the bad_opcode check was set above. I.e. exactly like the unknown opcodes above.

Of course, if you intend to replace all of these with functional variants before the hard fork, it doesn't matter.

Reply:
After activation this code can be removed. What prevents us from doing so is that the later check against having bad opcodes is gated by fExec. Under current consensus rules these opcodes terminate execution with a failure even inside of a non-executed if branch. That's one of the rules being relaxed by protocol cleanup.